nomos-c/src/api.rs

```diff
@@ -0,0 +1,86 @@
+use std::ffi::c_void;
+
+use nomos_node::RuntimeServiceId;
+use overwatch::overwatch::{Overwatch, OverwatchHandle};
+use tokio::runtime::{Handle, Runtime};
+
+use crate::NomosNodeErrorCode;
+
+// Define an opaque type for the complex Overwatch type
+type NomosOverwatch = Overwatch<RuntimeServiceId>;
+
+#[repr(C)]
+pub struct NomosNode {
+    // Use opaque pointer instead of the generic type
+    overwatch: *mut c_void,
+    // Keep simple types as-is
+    runtime: *mut c_void,
+}
+
+impl NomosNode {
+    pub fn new(overwatch: NomosOverwatch, runtime: Runtime) -> Self {
+        Self {
+            // Box the complex types and convert to opaque pointers
+            overwatch: Box::into_raw(Box::new(overwatch)).cast::<c_void>(),
+            runtime: Box::into_raw(Box::new(runtime)).cast::<c_void>(),
+        }
+    }
+
+    // Helper methods to safely access the inner types
+    #[must_use]
+    const fn get_overwatch_handle(&self) -> &OverwatchHandle<RuntimeServiceId> {
+        unsafe { self.overwatch.cast::<NomosOverwatch>().as_ref().unwrap() }.handle()
```

Comment:
Should we use `expect`s here? So something will be printed to stderr that we can work with?